modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/imageio/MaskOverviewProvider.java

```diff
@@ -787,16 +803,15 @@ public SpiHelper(
             }
         }
 
-        public boolean isMultidim() {
-            return isMultidim;
+        public SpiHelper(ReadingAccessObject readObject) throws IOException {
+            this.readingAccessObject = readObject;
```

Comment:
Why duplicate fields, if the readingAccessObject (ugly name btw) already has them and it's a field, best to remove the other 3 no?

Reply:
done. also renamed the base classes 
(see https://github.com/geosolutions-it/imageio-ext/pull/228/)